components/supervisor/pkg/serverapi/publicapi.go

```diff
@@ -319,24 +298,47 @@ func (s *Service) InstanceUpdates(ctx context.Context, instanceID string, worksp
 	if !s.usePublicAPI(ctx) && s.persistServerAPIChannelWhenStart(ctx) {
 		return s.gitpodService.InstanceUpdates(ctx, instanceID)
 	}
+
 	updateChan := make(chan *gitpod.WorkspaceInstance)
-	var latestInstance *gitpod.WorkspaceInstance
+
+	// var latestInstance *gitpod.WorkspaceInstance
+	// go func() {
+	// 	for {
+	// 		if ctx.Err() != nil {
+	// 			close(updateChan)
+	// 			break
+	// 		}
+	// 		if instance, err := s.getWorkspaceInfo(ctx, instanceID, workspaceID); err == nil {
+	// 			if reflect.DeepEqual(latestInstance, instance) {
+	// 				continue
+	// 			}
+	// 			latestInstance = instance
+	// 			updateChan <- instance
+	// 		}
+	// 		time.Sleep(1 * time.Second)
+	// 	}
+	// }()
+
+	service := v1.NewWorkspacesServiceClient(s.publicAPIConn)
+	resp, err := service.WorkspaceStatusUpdate(ctx, &v1.WorkspaceStatusUpdateRequest{
+		WorkspaceId: workspaceID,
+	})
+	if err != nil {
+		log.WithField("method", "WorkspaceStatusUpdate").WithError(err).Error("failed to call PublicAPI")
+		return nil, err
+	}
 	go func() {
```

Comment:
Given the method signature, there's no way for the caller to stop listening. This means it will listen forever (as long as the connection holds). Is that actually desirable?

Reply:
yeah I think this is expected, cc @mustard-mh @iQQBot if you have any input on this